lms/services/canvas_studio.py

```diff
@@ -0,0 +1,221 @@
+        self._client_id = application_instance.settings.get(
+            "canvas_studio", "client_id"
+        )
+        self._client_secret = application_instance.settings.get_secret(
+            request.find_service(AESService), "canvas_studio", "client_secret"
+        )
+        self._oauth_http_service = oauth_http_factory(
+            {}, request, service=Service.CANVAS_STUDIO
+        )
+        self._request = request
+
+    def get_access_token(self, code: str) -> None:
+        """
+        Fetch and persist an access token for Canvas Studio API calls.
+
+        :param code: Authorization code received from OAuth callback
+        """
+        token_url = self._api_url("oauth/token")
+        self._oauth_http_service.get_access_token(
+            token_url,
+            self.redirect_uri(),
+            auth=(self._client_id, self._client_secret),
+            authorization_code=code,
+        )
+
+    def authorization_url(self, state: str) -> str:
+        """
+        Construct the authorization endpoint URL for Canvas Studio.
+
+        This constructs the authorization URL for the Canvas Studio instance
+        associated with the current application instance.
+
+        :param state: `state` query param for the authorization request
+        """
+        auth_url = urlunparse(
+            (
+                "https",
+                self._domain,
+                "api/public/oauth/authorize",
+                "",
+                urlencode(
+                    {
+                        "client_id": self._client_id,
+                        "response_type": "code",
+                        "redirect_uri": self.redirect_uri(),
+                        "state": state,
+                    }
+                ),
+                "",
+            )
+        )
+
+        return auth_url
+
+    def redirect_uri(self) -> str:
+        """Return OAuth redirect URI for Canvas Studio."""
+        return self._request.route_url("canvas_studio_api.oauth.callback")
+
+    def list_media_library(self) -> list[File]:
+        """
+        List the videos and collections for the current user.
+
+        The result of this call corresponds to what the user sees if they
+        visit Canvas Studio from within their LMS, or use the Canvas Studio
+        picker when creating a Page.
+        """
+
+        collections = self._api_request("v1/collections", CanvasStudioCollectionsSchema)
+        user_collection = None
+
+        files = []
+        for collection in collections:
+            if collection["type"] == "user":
+                user_collection = collection
+                continue
+
+            files.append(
+                {
+                    "type": "Folder",
+                    "display_name": collection["name"],
+                    "updated_at": collection["created_at"],
+                    "id": str(collection["id"]),
+                    "contents": {
```

Comment:
Endpoint to request the nested level contents. This is similar to how BB files work.